healthcheck/http.go

```diff
@@ -209,13 +209,13 @@ func (h *HTTPHealthcheck) LogInfo(message string) {
 }
 
 // Execute executes an healthcheck on the given target
-func (h *HTTPHealthcheck) Execute() error {
+func (h *HTTPHealthcheck) Execute() ExecutionError {
```

Comment:
Would it not be better to implement "error" on ExecutionError and then, when the error is used, try to cast it to ExecutionError to extract the attributes? A function taking an error and returning the attributes (or an empty map in case the error isn't a ExecutionError) could even be used.